crates/circuit/src/dag_circuit.rs

```diff
@@ -5958,7 +5856,194 @@ impl DAGCircuit {
             .filter(|node| matches!(self.dag.node_weight(*node).unwrap(), NodeType::Operation(_)))
     }
 
-    fn substitute_node_with_subgraph(
+    /// Substitutes a node by a given DAGCircuit and adds the replacement DAG's global phase to the current DAG.
+    ///
+    /// # Arguments:
```

Comment:
I think Rust docs just treat this as a header, so no need for `:` here or in `Returns:`, but instead we'd use a linebreak 🙂 (see e.g. https://doc.rust-lang.org/rust-by-example/meta/doc.html)
```suggestion
    /// # Arguments
    ///
```

Reply:
Fixed in https://github.com/Qiskit/qiskit/pull/14766/commits/0c7d20d66b0a442271502472be10fd6a9b43dd65. Thx!